app/controllers/scan_analytics_controlller.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+class ScanAnalyticsController < ApplicationController
+  helper ProjectsHelper
+
+  before_action :set_project_or_fail
+  before_action :analytics_data, only: :index
+
+  def index
+    return if @analytics.blank?
+
+    params[:code_set_id] = params[:code_set_id].nil? ? @analytics.first.code_set_id : params[:code_set_id]
+    @scan_data = @analytics.where(code_set_id: params[:code_set_id]).first
+  end
+
+  def charts
+    return unless @project.coverity_project_id
+
+    scan_project_id = params[:scan_id].presence || @project.coverity_project_id
```

Comment:
I believe you do not need `.presence` here.

Reply:
params[:scan_id] it might be " " in params. SO handling that scenario is required.